tests/http/fixtures/agent_client.py

```
@@ -42,28 +43,57 @@ def post(
+        import socket
+        try:
+            # Parse host:port from base_url
+            from urllib.parse import urlparse
+            parsed = urlparse(self.base_url)
+            host = parsed.hostname or '127.0.0.1'
+            port = parsed.port or 80
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.settimeout(timeout)
+            result = sock.connect_ex((host, port))
+            sock.close()
+            return result == 0
```

Ruff (0.15.2)

[warning] Consider moving this statement to an `else` block

(TRY300)